### ISSUE TITLE
The default filters do not correctly recognize URLs containing get parameters.

### ISSUE THREAD

**Issue by lpe73** [downthemall/downthemall#193]
The default filters do not correctly recognize URLs containing get parameters.

For example when using the audio filter DTA fails to recognize
```
http://example.com/audio_file.mp3?some_get_parameter=1
```
but will download
```
http://example.com/document_file.pdf?some_get_parameter=1.mp3
```
when it shouldn't. The default regex filters should be changed to fix this.

**Comment by nmaier** [comment]
Where does this happen?

**Comment by msv6r** [comment]
This problem will occur for any link that contains a [query string](https://en.wikipedia.org/wiki/Query_string) when using the default filters.

To reproduce this bug create an HTML document containing the following links:

```
<a href="http://example.com/audio_file.mp3?some_get_parameter=1">audio file</a>

<a href="http://example.com/document_file.pdf?some_get_parameter=1.mp3">document file</a>
```

Notice that the default filters will misidentify these links because they are matching against the [query string](https://en.wikipedia.org/wiki/Query_string) rather than stopping at the end of the file name.

This problem can be solved by modifying the default regex filters to ignore everything after the "?" in the URL.

I have noticed that this problem also applies to [fragment identifiers](https://en.wikipedia.org/wiki/Fragment_identifier) for the same reasons and can be fixed by modifying the default regex filters to also ignore everything after the "#" in the URL.

**Comment by d7415** [comment]
I agree in principle (I've used custom filters to work around URLs like your first example in the past with e.g. `*.mp3?*`) but I'm not convinced it's that simple, because there are also cases in the form `http://example.com/download.pl?file=audio.mp3`.

For the defaults I don't personally mind, so long as the custom filters can support the full address, but there are a lot of edge cases out there. Widening the default filters may be useful though.

I'm not sure how best to handle your second example, but I'd suggest having strong words with whoever chose that scheme!

**Comment by nmaier** [comment]
What @d7415 said essentially regarding query strings.
Fragments shouldn't match, tho

**Comment by msv6r** [comment]
Given the example @d7415 provided I'm now thinking it would be best to have the default filters match against both the full URL and the URL excluding the query sting.
